### PR TITLE
Update Forms base SCSS with latest styles

### PR DESCRIPTION
**Pull request by josephco**
Updates Forms styles to be included in core_config_nightshade in Casper repo

### CODE REVIEW

File: src/forms/_base.scss [hunk 1]
```diff
@@ -173,30 +120,26 @@ select {
...
+  @include respond-at-and-above($form-max) {
     display: flex;
   }
 }
 
 .form-legend-alt {
-  @include body--sans-small;
-
+  font-size: ms(0);
+  letter-spacing: size(smaller-tracking);
```
Comment (ckeisato) [inline src/forms/_base.scss:137]: why do you need to state the letter spacing here?  it doesn't inherit an appropriate value from the body?

File: src/forms/_base.scss [hunk 2]
```diff
@@ -219,18 +158,77 @@ select {
...
+    top: ms(-4);
+    left: 1.1rem;
+    font-size: ms(-1);
+    letter-spacing: size(smaller-tracking);
+    line-height: 1;
+    color: color(dark-text);
+    opacity: 0;
+    pointer-events: none;
+    transform: translateY(ms(-9));
+    transition: transform 0.25s ease-out, opacity 0.25s ease-out;
```
Comment (ckeisato) [inline src/forms/_base.scss:226]: didn't we have a transition mixin?  or a transition variable?

File: src/forms/_base.scss [hunk 3]
```diff
@@ -345,17 +365,17 @@ select {
...
+  font-size: ms(-1);
+  color: color(primary);
+  margin-top: 0.7rem;
 
   a {
-    @include link(text, medium-text, "after");
+    @include link(text, dark-text, "after");
+  }
+
+  &.form-help--tight {
```
Comment (ckeisato) [inline src/forms/_base.scss:377]: what is the `--tight` version for?

File: src/forms/_base.scss (hunk 2 above)
Comment (josephco) [inline src/forms/_base.scss:226]: Not sure, we usually just write it out like this.

File: src/forms/_base.scss (hunk 3 above)
Comment (josephco) [inline src/forms/_base.scss:377]: Not sure, this was carried over from casper repo, prob where there needed to be less margin in some instance.

File: src/forms/_base.scss [hunk 4]
```diff
@@ -173,30 +120,26 @@ select {
...
     display: flex;
   }
 }
 
 .form-legend-alt {
-  @include body--sans-small;
-
+  font-size: ms(0);
+  letter-spacing: size(smaller-tracking);
   text-transform: lowercase;
```
Comment (ckeisato) [inline src/forms/_base.scss:138]: why is this necessary?

File: src/forms/_base.scss (hunk 1 above)
Comment (josephco) [inline src/forms/_base.scss:137]: This is overwriting styles because it's nested within an H2

**Comment by ckeisato** [comment]
looks good, is there a corresponding forms ie stylesheet that should be updated?  just noticed you added some flexbox stuff.

File: src/forms/_base.scss (hunk 4 above)
Comment (josephco) [inline src/forms/_base.scss:138]: Overwrites H2 styles
